src/async_impl/request.rs

```diff
@@ -504,7 +504,7 @@ impl RequestBuilder {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn send(self) -> impl Future<Output = Result<Response, crate::Error>> {
+    pub fn send(self) -> impl Future<Output = Result<Response, crate::Error>> + Sync {
```

Comment:
IMHO you could also put the `Send` bound as well

```suggestion
    pub fn send(self) -> impl Future<Output = Result<Response, crate::Error>> + Send + Sync {
```